src/scripts/templates/gift_card.js

```diff
@@ -1,32 +1,35 @@
+import QRCode from 'qrcode';
+
 /**
  * Gift Card Template Script
  * ------------------------------------------------------------------------------
  * A file that contains scripts highly couple code to the Gift Card template.
  */
 
-import $ from 'jquery';
-
-const config = {
-  qrCode: '#QrCode',
-  printButton: '#PrintGiftCard',
-  giftCardCode: '.giftcard__code',
+const selectors = {
+  qrCodeCanvas: '[data-gift-card-qr]',
+  printButton: '[data-gift-card-print]',
+  giftCardCode: '[data-gift-card-digits]',
 };
 
-// new QRCode($qrCode[0], {
-//   text: $qrCode.attr('data-identifier'),
-//   width: 120,
-//   height: 120,
-// });
+// This is the QR code that allows customers to use at a POS
+document.querySelectorAll(selectors.qrCodeCanvas).forEach((qrCodeCanvas) => {
+  QRCode.toCanvas(qrCodeCanvas, qrCodeCanvas.getAttribute('data-identifier'));
+});
 
-$(config.printButton).on('click', () => {
-  window.print();
+document.querySelectorAll(selectors.printButton).forEach((printButton) => {
+  printButton.addEventListener('click', () => {
+    window.print();
+  });
 });
 
 // Auto-select gift card code on click, based on ID passed to the function
-$(config.giftCardCode).on('click', {element: 'GiftCardDigits'}, selectText);
+document.querySelectorAll(selectors.giftCardCode).forEach((giftCardCode) => {
+  giftCardCode.addEventListener('click', selectText);
+});
 
 function selectText(evt) {
-  const text = document.getElementById(evt.data.element);
+  const text = evt.target;
   let range = '';
 
   if (document.body.createTextRange) {
```

Comment:
Looks like `document.body.createTextRange` is an old IE only method. `window.getSelection` is supported on all the browsers themes currently support (IE11 + evergreen browsers), so we can simplify this method and remove this condition.